tests/platform_tests/cli/test_show_platform.py

```diff
@@ -101,18 +102,49 @@ def test_show_platform_psustatus(duthosts, rand_one_dut_hostname):
     """
     duthost = duthosts[rand_one_dut_hostname]
     logging.info("Check pmon daemon status")
-    assert check_pmon_daemon_status(duthost), "Not all pmon daemons running."
+    pytest_assert(check_pmon_daemon_status(duthost), "Not all pmon daemons running.")
 
     cmd = " ".join([CMD_SHOW_PLATFORM, "psustatus"])
 
     logging.info("Verifying output of '{}' ...".format(cmd))
     psu_status_output_lines = duthost.command(cmd)["stdout_lines"]
-    psu_line_pattern = re.compile(r"PSU\s+\d+\s+(OK|NOT OK|NOT PRESENT)")
+
+    if "201811" in duthost.os_version or "201911" in duthost.os_version:
+        psu_line_pattern = re.compile(r"PSU\s+\d+\s+(OK|NOT OK|NOT PRESENT)")
+    else:
+        psu_line_pattern = re.compile(r"PSU\s+\d+\s+\w+\s+\w+\s+\w+\s+\w+\s+\w+\s+(OK|NOT OK|NOT PRESENT)\s+(green|amber|red|off)")
+
     for line in psu_status_output_lines[2:]:
         pytest_assert(psu_line_pattern.match(line), "Unexpected PSU status output: '{}'".format(line))
         # TODO: Compare against expected platform-specific output
 
 
+def test_show_platform_psustatus_json(duthosts, rand_one_dut_hostname):
+    """
+    @summary: Verify output of `show platform psustatus --json`
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+
+    if "201811" in duthost.os_version or "201911" in duthost.os_version:
+        pytest.skip("JSON output not available in this version")
+
+    logging.info("Check pmon daemon status")
+    pytest_assert(check_pmon_daemon_status(duthost), "Not all pmon daemons running.")
+
+    cmd = " ".join([CMD_SHOW_PLATFORM, "psustatus", "--json"])
+
+    logging.info("Verifying output of '{}' ...".format(cmd))
+    psu_status_output = duthost.command(cmd)["stdout"]
+    psu_info_list = json.loads(psu_status_output)
+
+    # TODO: Compare against expected platform-specific output
+    for psu_info in psu_info_list:
+        expected_keys = ["index", "name", "presence", "status", "led_status", "model", "serial", "voltage", "current", "power"]
+        pytest_assert(all(key in psu_info for key in expected_keys), "Unexpected PSU status JSON output: '{}'".format(psu_status_output))
```

Comment:
The error message should be updated too.

Reply:
Rephrased. Please check.